File: src/main/java/leets/weeth/domain/schedule/application/validator/ScheduleTimeCheckValidator.java
package leets.weeth.domain.schedule.application.validator;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;
import leets.weeth.domain.schedule.application.annotation.ScheduleTimeCheck;
import leets.weeth.domain.schedule.application.dto.ScheduleDTO.Time;

public class ScheduleTimeCheckValidator implements ConstraintValidator<ScheduleTimeCheck, Time> {

    @Override
    public void initialize(ScheduleTimeCheck constraintAnnotation) {
        ConstraintValidator.super.initialize(constraintAnnotation);
    }

    @Override
    public boolean isValid(Time time, ConstraintValidatorContext context) {
        return time.start().isBefore(time.end().plusMinutes(1));
    }
}

File: src/main/kotlin/com/weeth/domain/schedule/application/validator/ScheduleTimeCheckValidator.kt
package com.weeth.domain.schedule.application.validator

import com.weeth.domain.schedule.application.annotation.ScheduleTimeCheck
import com.weeth.domain.schedule.application.dto.request.ScheduleTimeRequest
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

/**
 * Todo: 사용처 있는지 확인하고 없으면 제거
 */
class ScheduleTimeCheckValidator : ConstraintValidator<ScheduleTimeCheck, ScheduleTimeRequest> {
    override fun isValid(
        time: ScheduleTimeRequest?,
        context: ConstraintValidatorContext,
    ): Boolean = time == null || time.start.isBefore(time.end.plusMinutes(1))
}
